duckdb_engine/tests/test_basic.py

```diff
@@ -408,3 +410,11 @@ def test_do_ping(tmp_path: Path, caplog: LogCaptureFixture) -> None:
         assert any(
             "Pool pre-ping on connection" in message for message in caplog.messages
         )
+
+
+def test_try_cast(engine: Engine) -> None:
+    try_cast = importorskip("sqlalchemy", "2.0.14").try_cast
+
+    with engine.connect() as conn:
+        query = select(try_cast("2022-01-01", DateTime))
```

Comment:
Could add a try cast that results in null:

```python
        query = select(try_cast("not a date", DateTime))
        assert conn.execute(query).one() == (None, ) # not quite sure what will result
```

though I sorta doubt that this would find any additional bugs that this existing test wouldn't find. But seems easy enough to add. I'd be fine keeping this as it is too.